File: BookNestFrontend/booknest_desktop/lib/models/reservation_status.dart
class ReservationStatus {
  final int id;
  final String name;

  ReservationStatus({required this.id, required this.name});

  factory ReservationStatus.fromJson(Map<String, dynamic> json) {
    return ReservationStatus(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
    );
  }
}

File: BookNestBackend/BookNest.Services/Database/Entities/ReservationStatus.cs
using System.ComponentModel.DataAnnotations;

namespace BookNest.Services.Database.Entities
{
    public class ReservationStatus
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        public ICollection<EventReservation> EventReservations { get; set; } = new List<EventReservation>();
    }
}
